vignettes/workflows.Rmd

```diff
@@ -0,0 +1,292 @@
+
+
+```
+
+
+With information about tokens in hand, we can start asking questions. For instance, how does the relative frequency of words relate to their position in the turn?
+
+To explore this question, let's look at a shorter excerpt: 1 minute in total, divided over 4 lines. To do this, we create a dataframe `this_conv`, dividing it into 4 lines of 15 seconds each. We also create a dataframe `these_tokens` with tokenized turn elements for the same conversation, divided up in the same way.
+
+```{r}
+
+this_conv <- conv |>
+  add_lines(line_duration=15000) |>
+  dplyr::filter(source == "/dutch2/DVA12S",
+                line_id < 5) # let's look at the first three lines
+
+these_tokens <- conv_tokens |>
+  add_lines(line_duration=15000, time_columns = "relative_time") |>
+  dplyr::filter(source == "/dutch2/DVA12S",
+                line_id < 5)
+
+this_conv |>
+  ggplot(aes(x = line_end, y = line_participant)) +
+  ggtitle("Relative frequency of elements within turns") +
+  scale_y_reverse() + # we reverse the axis because lines run top to bottom
+  geom_turn(aes(
+    begin = line_begin,
+    end = line_end)) +
+  geom_token(data=these_tokens,
+             aes(x=line_relative_time,
+                 size=frequency)) +
+  xlab("Time (ms)") +
+  ylab("") +
+  theme_turnPlot()
+
+p <- last_plot()
+
+```
+
+
+Finally, we can also print the content of some of the elements. Here, we pick the most frequent turn-initial elements for plotting, highlight them with another layer of `geom_token()` and plot the text using `geom_label_repel()`:
+
+```{r}
+
+these_tokens_first <- these_tokens |>
+  dplyr::filter(order=="first",
+                rank < 10)
+
+p +
+  ggtitle("Some frequent turn-initial elements") +
+  geom_token(data=these_tokens_first,
+             aes(x=line_relative_time),
+             color="red") +
+  ggrepel::geom_label_repel(data=these_tokens_first,
+                            aes(x=line_relative_time,
+                                label=token),
+                            direction="y")
+
+```
+
+
+
+
+### Notes 
+
+Token frequencies are calculated over the entire dataset. If you want source-specific data, you can filter the source prior to tokenization:
+
+```r
+tokens_DVA9M <- data |>
+  dplyr::filter(source == "/dutch2/DVA9M") |>
+  tokenize()
+
+tokens_DVA9M
+```
+
+## References
+
+* Liesenfeld, Andreas, and Mark Dingemanse. 2022. ‘Building and Curating Conversational Corpora for Diversity-Aware Language Science and Technology’. In Proceedings of the 13th Conference on Language Resources and Evaluation (LREC 2022), 1178–92. Marseille. doi:10.48550/arXiv.2203.03399.
+* Stivers, Tanya, N. J. Enfield, Penelope Brown, C. Englert, Makoto Hayashi, Trine Heinemann, Gertie Hoymann, Federico Rossano, J. P. de Ruiter, Kyung-Eun Yoon, and Stephen C. Levinson. 2009. ‘Universals and Cultural Variation in Turn-Taking in Conversation’. Proceedings of the National Academy of Sciences 106 (26): 10587–92. doi:10.1073/pnas.0903616106.
```

Comment:
Given that we are citing these papers here, we should consider putting them in the Description, too. CRAN has specific requirements for citations in the description though, so let's keep that in mind:

> - Relevant citations should be included in the description. These should be in author-year style, preferably followed by an identifier such as DOI, arXiv id, or ISBN for published materials.
> - DOIs should be enclosed in angle brackets, and formatted as doi:10.prefix/suffix. Example: Sugihara (1994) doi:10.1098/rsta.1994.0106.

Reply:
Our own paper, maybe, but the other backs up a point about the expected distribution